### ISSUE TITLE
Getting Started Docs Failing

### ISSUE THREAD

**Issue by thesefer** [Icinga/ansible-collection-icinga#320]
Trying to setup a new full Icinga2 installation on a host which will be the first Master using: https://github.com/Icinga/ansible-collection-icinga/blob/main/doc/getting-started.md
-> Example Playbooks -> "This is a example on how to install Icinga 2 server with Icinga Web 2 and Icinga DB."

I couldn't find newer playbooks / examples so I used the one found within the `docs` directory.

Collection 0.3.4 Error
```
TASK [icinga.icinga.icinga2 : collect all config objects in play vars tmp_objects={{ tmp_objects| default([]) + lookup('list', icinga2_objects) }}] ***************************************************************************************************************************************************************************
fatal: [tlp87.net]: FAILED! => {"msg": "Unexpected templating type error occurred on ({{ tmp_objects| default([]) + lookup('list', icinga2_objects) }}): can only concatenate list (not \"dict\") to list"}
```

Collection 0.3.3 Error
```
RUNNING HANDLER [icinga.icinga.icingadb_redis : icingadb-redis-restart name={{ icingadb_redis_service_name }}, state=restarted] ***********************************************************************************************************************************************************************************************
fatal: [tlp87.net]: FAILED! => {"changed": false, "msg": "Unable to restart service icingadb-redis: Job for icingadb-redis.service failed because the control process exited with error code.\nSee \"systemctl status icingadb-redis.service\" and \"journalctl -xeu icingadb-redis.service\" for details.\n"}
```

Output from icingadb-redis
```
× icingadb-redis.service - Redis database for Icinga DB
     Loaded: loaded (/lib/systemd/system/icingadb-redis.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Mon 2024-08-12 15:19:06 CEST; 40s ago
    Process: 167483 ExecStart=/usr/bin/icingadb-redis-server /usr/share/icingadb-redis/icingadb-redis-systemd.conf (code=exited, status=1/FAILURE)
   Main PID: 167483 (code=exited, status=1/FAILURE)
        CPU: 49ms

Aug 12 15:19:06 tlp87 systemd[1]: Starting Redis database for Icinga DB...
Aug 12 15:19:06 tlp87 icingadb-redis-server[167483]: *** FATAL CONFIG FILE ERROR (Redis 7.0.15) ***
Aug 12 15:19:06 tlp87 icingadb-redis-server[167483]: Can't open the log file: No such file or directory
Aug 12 15:19:06 tlp87 systemd[1]: icingadb-redis.service: Main process exited, code=exited, status=1/FAILURE
Aug 12 15:19:06 tlp87 systemd[1]: icingadb-redis.service: Failed with result 'exit-code'.
Aug 12 15:19:06 tlp87 systemd[1]: Failed to start Redis database for Icinga DB.
```

`/usr/share/icingadb-redis/icingadb-redis-systemd.conf` -> `/etc/icingadb-redis/icingadb-redis.conf`
Log line: `logfile /var/log/icingadb-redis/redis.log`
No folder: `ls: cannot access '/var/log/icingadb-redis/': No such file or directory`

---

Playbook:
```
- hosts: tlp87.net
  become: yes
  roles:
    - int.env
    - int.apt
    # - { role: int.certbot, tags: ['certbot'] }
    - { role: geerlingguy.mysql, tags: ['mysql'] }
    - icinga.icinga.repos
    - icinga.icinga.icinga2
    - icinga.icinga.icingadb
    - icinga.icinga.icingadb_redis
    - icinga.icinga.monitoring_plugins
    - icinga.icinga.icingaweb2
```

Vars:
```
# Variables for collection icinga
icinga_repo_apt_stable_deb: "deb [signed-by={{ icinga_repo_apt_keyring }}] https://artifactory.net/artifactory/icinga-generic-remote/{{ ansible_distribution|lower }} icinga-{{ ansible_distribution_release|lower }} main"
icinga_repo_apt_testing_deb: "deb [signed-by={{ icinga_repo_apt_keyring }}] https://artifactory.net/artifactory/icinga-generic-remote/{{ ansible_distribution|lower }} icinga-{{ ansible_distribution_release|lower }}-testing main"
icinga_repo_apt_snapshot_deb: "deb [signed-by={{ icinga_repo_apt_keyring }}] https://artifactory.net/artifactory/icinga-generic-remote/{{ ansible_distribution|lower }} icinga-{{ ansible_distribution_release|lower }}-snapshots main"
icinga_repo_gpgkey: "https://artifactory.net/artifactory/icinga-generic-remote/icinga.key"
icinga_repo_stable: true
icinga_repo_testing: false
icinga_repo_snapshot: false

icinga_monitoring_plugins_check_commands:
    - "all"

icingaweb2_db:
  type: mysql
  name: icingaweb
  host: 127.0.0.1
  user: icingaweb
  password: abcd
icingaweb2_db_import_schema: true

icingadb_database_type: mysql
icingadb_database_host: localhost
icingadb_database_user: icingadb
icingadb_database_password: efgh
icingadb_database_import_schema: true

icinga2_constants:  # Set default constants and TicketSalt for the CA
  TicketSalt: "{{ lookup('ansible.builtin.password', '.icinga-server-ticketsalt') }}"
  NodeName: "{{ ansible_fqdn }}"
  ZoneName: "main"
icinga2_confd: false # Disable example configuration
icinga2_purge_features: yes # Ansible will manage all features
icinga2_config_directories:  # List of directories in which the role will manage monitoring objects
  - zones.d/main/commands
  - zones.d/main/hosts
  - zones.d/main/services
icinga2_features:
  - name: icingadb
    host: 127.0.0.1
  - name: notification
  - name: checker
  - name: mainlog
  - name: api
    ca_host: none
    endpoints:
      - name: "{{ ansible_fqdn }}"
    zones:
      - name: "main"
        endpoints:
          - "{{ ansible_fqdn }}"
icinga2_objects:
  '{{ ansible_fqdn }}':
    - name: root
      type: ApiUser
      file: conf.d/apiusers.conf
      password: ijkl
      permissions:
        - '*'

# Variables for geerlingguy.mysql
mysql_innodb_file_format: barracuda
mysql_innodb_large_prefix: 1
mysql_innodb_file_per_table: 1
mysql_packages:
  - mariadb-client
  - mariadb-server
  - python3-mysqldb
mysql_users:
  - name: icingaweb
    host: "%"
    password: "{{ icingaweb2_db['password'] }}"
    priv: "icingaweb.*:ALL"
  - name: icingadb
    host: "%"
    password: "{{ icingadb_database_password }}"
    priv: "icingadb.*:ALL"
mysql_databases:
  - name: icingadb
  - name: icingaweb
  - name: vspheredb
    encoding: utf8mb4
    collation: utf8mb4_unicode_ci
  - name: director
```

**Comment by thesefer** [comment]
Hi
thanks for your quick response! Could have sworn this is possible with ansible but I guess I confused it with another situation. With 0.3.3 the playvar worked perfectly fine.

I tried it again with 0.3.4., same error
`fatal: [tlp87.lp.rsint.net]: FAILED! => {"msg": "Unexpected templating type error occurred on ({{ tmp_objects| default([]) + lookup('list', icinga2_objects) }}): can only concatenate list (not \"dict\") to list"}`

with
``` 
icinga2_objects:
  tlp87.net:
    - name: root
      type: ApiUser
      file: conf.d/apiusers.conf
      password: ikjl
      permissions:
        - '*'
```

**Comment by mkayontour** [comment]
Did you configure icinga2_objects somewhere else again? Or is it the only occurrence? 
Can you define the var in a host_vars? 
This is a bug, you can only define the objects variable with the fqdn as Key in host_vars. If outside of the host_vars we expect the other syntax. 

```
icinga2_objects:
  - name: root
    object_type: ApiUser
    ... 
```

**Comment by thesefer** [comment]
It is not configured elsewhere. As per my opener I am working on a completely fresh setup (on a newly spawned vm). 

The vars provided reside in `host_vars/tlp87.net` which is a symlink to `inventory/host_vars` and I invoke the play via `ansible-playbook playbooks/site.yml --limit tlp87.net`.

**Comment by thesefer** [comment]
@mkayontour I installed 9a0e874800e093fcdc57e46b29c8e70326e9999e for a test and it looks like the issue has been fixed (as per https://github.com/Icinga/ansible-collection-icinga/pull/311 I think, no error anymore).

But unfortunately the task is being skipped for the host and the object is not being deployed. Probably the reason I do not get an error :)

**Comment by Donien** [comment]
Hi @thesefer 

as per  
[https://github.com/Icinga/ansible-collection-icinga/blob/9a0e874800e093fcdc57e46b29c8e70326e9999e/roles/icinga2/defaults/main.yml#L18](https://github.com/Icinga/ansible-collection-icinga/blob/9a0e874800e093fcdc57e46b29c8e70326e9999e/roles/icinga2/defaults/main.yml#L18)  
you can actually use `inventory_hostname`.

This variable is basically meant for use cases like yours.  
If you change it to `inventory_hostname`, your situation could look like this:

Inventory:

```ini
[icingahosts]
icinga-master
; actual fqdn e.g. being "master.example.com"
```

(Host) Vars:

```yaml
icinga2_config_host: "{{ inventory_hostname }}"
icinga2_objects:
  icinga-master:
    - name: root
      type: ApiUser
      file: conf.d/apiusers.conf
      password: ijkl
      permissions:
        - '*'
```

**Comment by thesefer** [comment]
Hi @Donien 

Thanks for your explanation. You are right, I could use this variable! Maybe I have overlooked it in the documentation, I searched for it due to the output log.

> Oftentimes - at least from what I've seen - the name in your inventory is the FQDN of the host.
So using the FQDN for icinga2_config_host seems like a reasonable default to me.

This is what I am doing, too. In my inventory the host looks like this, i.e.:

```
[test_services]
tlp87.intern.example.net ansible_host=192.168.22.10
```

So the `inventory_hostname` would be correct. But the `ansible_fqdn` is gathered from the system and the hostname is `tlp87` only.

It's interesting how different things are. I have never seen a FQDN as the hostname on a machine. Instead I always see `inventory_hostname_short` in this case.

Thank you very much for your help and support! I will continue testing everything now :-)
